### ISSUE TITLE
updating addon throws `TypeError: this.worker is null` for each app

### ISSUE THREAD

**Issue by mykmelez** [mozilla/r2d2b2g#169]
Upon updating from 0.8 to 1.0rc1 with two apps, the Simulator threw `TypeError: this.worker is null` for each app:

```
info: r2d2b2g: Simulator.updateApp http://areatweet.com/
info: r2d2b2g: Creating webapp entry: {
  "origin": "http://areatweet.com",
  "installOrigin": "http://areatweet.com",
  "receipt": null,
  "installTime": 1354386153538,
  "appStatus": 1,
  "localId": 44,
  "manifestURL": "http://areatweet.com/"
}
info: r2d2b2g: Simulator.updateApp /home/myk/Mozilla/gaia/test_apps/uitest/manifest.webapp
info: r2d2b2g: Creating webapp entry: {
  "origin": "app://myapp45.gaiamobile.org",
  "installOrigin": "app://myapp45.gaiamobile.org",
  "receipt": null,
  "installTime": 1354386153542,
  "appStatus": 3,
  "localId": 45,
  "manifestURL": "app://myapp45.gaiamobile.org/manifest.webapp"
}
info: r2d2b2g: Created /home/myk/.mozilla/firefox/do32rbri.default/extensions/r2d2b2g@mozilla.org/profile/webapps/myapp44.gaiamobile.org
info: r2d2b2g: Simulator.sendListApps
error: r2d2b2g: An exception occurred.
Traceback (most recent call last):
  File "resource://gre/modules/NetUtil.jsm", line 85, in 
    aCallback(aStatusCode);
  File "resource://r2d2b2g-at-mozilla-dot-org/api-utils/lib/text-streams.js", line 185, in TextWriter_writeAsync/<
    callback.call(self, err);
  File "resource://r2d2b2g-at-mozilla-dot-org/r2d2b2g/lib/main.js", line 238, in simulator.updateApp/<
    simulator.sendListApps();
  File "resource://r2d2b2g-at-mozilla-dot-org/r2d2b2g/lib/main.js", line 476, in simulator.sendListApps
    this.worker.postMessage({
TypeError: this.worker is null
info: r2d2b2g: Created /home/myk/.mozilla/firefox/do32rbri.default/extensions/r2d2b2g@mozilla.org/profile/webapps/myapp45.gaiamobile.org
info: r2d2b2g: Zipping /home/myk/Mozilla/gaia/test_apps/uitest to /home/myk/.mozilla/firefox/do32rbri.default/extensions/r2d2b2g@mozilla.org/profile/webapps/myapp45.gaiamobile.org/application.zip
info: r2d2b2g: archived dir /home/myk/Mozilla/gaia/test_apps/uitest
info: r2d2b2g: Simulator.sendListApps
error: r2d2b2g: An exception occurred.
Traceback (most recent call last):
  File "resource://gre/modules/NetUtil.jsm", line 85, in 
    aCallback(aStatusCode);
  File "resource://r2d2b2g-at-mozilla-dot-org/api-utils/lib/text-streams.js", line 185, in TextWriter_writeAsync/<
    callback.call(self, err);
  File "resource://r2d2b2g-at-mozilla-dot-org/r2d2b2g/lib/main.js", line 238, in simulator.updateApp/<
    simulator.sendListApps();
  File "resource://r2d2b2g-at-mozilla-dot-org/r2d2b2g/lib/main.js", line 476, in simulator.sendListApps
    this.worker.postMessage({
TypeError: this.worker is null
```

The Manager was not open at the time. @digitarald: perhaps the code assumes it is?

**Comment by mykmelez** [comment]
Kris Maglione reports getting this the second time he tried removing a packaged app. He also says:

> The next time I reloaded the Dashboard, no apps were listed. The next reload after that the app that I tried to remove was gone and the others were back.
> 
> Also, at this point, all sorts of strange things started happening with the Running/Stopped indicator. The text was gray and it didn't update when I closed the emulator window. After that, it still said running. Clicking it switched it to Stopped but launched a new emulator instance. Clicking it again switched it to Running and killed the emulator instance.
> 
> I easily reproduced this by adding one remote app and two packaged apps, removing one, restarting the emulator, refreshing the Dashboard, and attempting to remove a second app. I haven't tried to find a simplified STR.

**Comment by mykmelez** [comment]
I landed a fix for this in pull #266 that made it into the latest preview release, since which updating the addon no longer results in this exception being thrown.
